packages/ai-core/src/common/prompt-service.ts

```diff
@@ -181,13 +181,18 @@ export class PromptServiceImpl implements PromptService {
     getDefaultRawPrompt(id: string): PromptTemplate | undefined {
         return this._prompts[id];
     }
+
+    matchVariables(template: string): RegExpMatchArray[] {
+        return matchVariablesRegEx(template);
+    }
+
     async getPrompt(id: string, args?: { [key: string]: unknown }): Promise<ResolvedPromptTemplate | undefined> {
         const prompt = this.getRawPrompt(id);
         if (prompt === undefined) {
             return undefined;
         }
 
-        const matches = [...prompt.template.matchAll(PROMPT_VARIABLE_REGEX)];
+        const matches = this.matchVariables(prompt.template);
```

Comment:
Is there a reason why we extract this call of `matchVariablesRegEx()` into a method `this.matchVariables()`? Is this to simplify overriding it by adopters? Not sure this is a very common use case though.

I personally would keep this function call direct and not extract it into a separate method in this class (similar to how we do it in `agent-configuration-widget.tsx`.

If you like to keep the method (`this.matchVariables()`), I wouldn't make it `public` but rather `protected`; if we really want to keep the method at all.

Reply:
Should we also put the other invocation of `prompt.template.matchAll` into a function for consistency?
See `[...prompt.template.matchAll(PROMPT_FUNCTION_REGEX)]` on line 215.